main.py

```diff
@@ -51,9 +54,14 @@
 
   # Setup
   args = parser.parse_args()
+  os.system('mkdir -p results')
+  os.system('mkdir -p results/'+args.name)
```

Comment:
Rather than adding file strings together, use `os.path.join('results', args.name)` for more safety/cross-platform support. Same for when using the CSV writer.